imxrt-hal/src/can/mod.rs

```diff
@@ -0,0 +1,1114 @@
+        self.set_ccm_ccg();
+        self.set_tx();
+        self.set_rx();
+
+        ral::modify_reg!(ral::can, self.reg, MCR, MDIS: MDIS_0);
+
+        self.enter_freeze_mode();
+
+        ral::modify_reg!(ral::can, self.reg, CTRL1, LOM: LOM_1);
+        ral::modify_reg!(ral::can, self.reg, MCR, FRZ: FRZ_1);
+        while ral::read_reg!(ral::can, self.reg, MCR, LPMACK == LPMACK_1) {}
+
+        self.soft_reset();
+
+        while ral::read_reg!(ral::can, self.reg, MCR, FRZACK != FRZACK_1) {}
+
+        ral::modify_reg!(
+            ral::can,
+            self.reg,
+            MCR,
+            SRXDIS: SRXDIS_1,
+            IRMQ: IRMQ_1,
+            AEN: AEN_1,
+            LPRIOEN: LPRIOEN_1,
+            SLFWAK: SLFWAK_1,
+            WAKSRC: WAKSRC_1,
+            WRNEN: WRNEN_1,
+            WAKMSK: WAKMSK_1
+        );
+
+        ral::modify_reg!(ral::can, self.reg, MCR, |reg| reg & !0x8800);
+
+        ral::modify_reg!(
+            ral::can,
+            self.reg,
+            CTRL2,
+            RRS: RRS_1,
+            EACEN: EACEN_1,
+            MRP: MRP_1
+        );
+
+        self.disable_fifo();
+        self.exit_freeze_mode();
+    }
+
+    pub fn instance_number(&self) -> usize {
+        M::USIZE
+    }
+
+    pub fn is_can1(&self) -> bool {
+        M::USIZE == 1
+    }
+
+    pub fn is_can2(&self) -> bool {
+        M::USIZE == 2
+    }
+
+    pub fn base_address(&self) -> u32 {
+        match self.is_can1() {
+            true => CAN1_ADDR,
+            false => CAN2_ADDR,
+        }
+    }
+
+    pub fn free(self) -> ral::can::Instance {
+        self.reg
+    }
+
+    fn soft_reset(&mut self) {
+        ral::modify_reg!(ral::can, self.reg, MCR, SOFTRST: SOFTRST_1);
+        while ral::read_reg!(ral::can, self.reg, MCR, SOFTRST == SOFTRST_1) {}
+    }
+
+    fn enter_freeze_mode(&mut self) {
+        ral::modify_reg!(ral::can, self.reg, MCR, FRZ: FRZ_1);
+        ral::modify_reg!(ral::can, self.reg, MCR, HALT: HALT_1);
+        while ral::read_reg!(ral::can, self.reg, MCR, FRZACK != FRZACK_1) {}
+    }
+
+    fn exit_freeze_mode(&mut self) {
+        ral::modify_reg!(ral::can, self.reg, MCR, HALT: HALT_0);
+        while ral::read_reg!(ral::can, self.reg, MCR, FRZACK != FRZACK_0) {}
+    }
+
+    pub fn set_mrp(&mut self, mrp: bool) {
+        ral::modify_reg!(ral::can, self.reg, CTRL2, MRP: mrp as u32)
+    }
+
+    pub fn set_rrs(&mut self, rrs: bool) {
+        ral::modify_reg!(ral::can, self.reg, CTRL2, RRS: rrs as u32)
+    }
+
+    fn set_ccm_ccg(&mut self) {
+        match self.instance_number() {
+            1 => {
+                unsafe { modify_reg!(ral::ccm, CCM, CCGR0, |reg| reg | 0x3C000) };
+            }
+            2 => {
+                unsafe { modify_reg!(ral::ccm, CCM, CCGR0, |reg| reg | 0x3C0000) };
+            }
+            u => {
+                log::error!("Invalid Can instance (set_ccm_ccg): {:?}", u);
+            }
+        }
+    }
+
+    fn set_tx(&mut self) {
```

Comment:
Agreed that it would be preferable to stick to the convention: see this fork of [`imxrt-iomuxc`](https://github.com/imxrt-rs/imxrt-iomuxc/compare/master...dstric-aqueduct:imxrt-iomuxc:dev/can) for an implementation of the Pin trait for the CAN1/2 periphs on the 1060.

Reply:
Commit [`b4eca9f`](https://github.com/imxrt-rs/imxrt-hal/pull/122/commits/b4eca9fedca725de1902de3187b6d0d9166cb453) adopts the convention above.